src/Value.zig

```diff
@@ -2222,7 +2222,7 @@ pub fn ptrField(parent_ptr: Value, field_idx: u32, pt: Zcu.PerThread) !Value {
     const aggregate_ty = parent_ptr_ty.childType(zcu);
 
     const parent_ptr_info = parent_ptr_ty.ptrInfo(zcu);
-    assert(parent_ptr_info.flags.size == .one);
+    assert(parent_ptr_info.flags.size == .one or parent_ptr_info.flags.size == .c);
```

Comment:
Ugh, I hate C pointers. This should work fine, but please could you reflect it in the doc comment above?

Also, the same change (to the `assert` and the doc comment) is necessary in `ptrOptPayload` if you wouldn't mind doing that while you're at it. Behavior test case which fails now and will pass after that:
```zig
test "comptime-known C pointer to optional pointer" {
    const opt: ?*u8 = @ptrFromInt(0x1000);
    const outer_ptr: [*c]const ?*u8 = &opt;
    const inner_ptr = &outer_ptr.*.?;
    comptime assert(@TypeOf(inner_ptr) == [*c]const *u8);
    comptime assert(@intFromPtr(inner_ptr.*) == 0x1000);
}
```